compiler/rustc_builtin_macros/src/source_util.rs

```diff
@@ -210,8 +210,34 @@ pub(crate) fn expand_include_str(
                 MacEager::expr(cx.expr_str(cx.with_def_site_ctxt(bsp), interned_src))
```

Comment:
maybe worth looking into unifying the logic here and the `rustc_parse` logic somehow?

Reply:
I moved the logic to a separate function and called it from both places.